File: examples/csharp/SetupsWithinTest.cs

    [TestClass]
    public class TestClass
    {
        private readonly Mock<ISomeDependency> _mock = new Mock<ISomeDependency>();

        [TestMethod]
        public void Test1()
        {
            _mock.Setup(m => m.SomeMethod()).Returns(true);

            // Act
            bool r = doSomething();

            // Assert
            Assert.AreEqual(result, r);
        }

        [TestMethod]
        public void Test2()
        {
            _mock.Setup(m => m.SomeMethod()).Returns(false);

            // Act
            bool r = doSomething();

            // Assert
            Assert.AreEqual(result, r);
        }
    }

File: examples/dart/setups_within_test.dart
import 'package:mockito/mockito.dart';
import 'package:test/test.dart';

class ISomeDependency {
  bool someMethod() => throw UnimplementedError();
}

class MockISomeDependency extends Mock implements ISomeDependency {}

void main() {
  final _mock = MockISomeDependency();

  test('Test1', () {
    when(_mock.someMethod()).thenReturn(true);

    // Act
    bool r = doSomething();

    // Assert
    expect(r, true);
  });

  test('Test2', () {
    when(_mock.someMethod()).thenReturn(false);

    // Act
    bool r = doSomething();

    // Assert
    expect(r, false);
  });
}

bool doSomething() {
  return true;
}
